pkg/auth/local_authenticator_test.go

```diff
@@ -0,0 +1,41 @@
+package auth
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+var _ = Describe("AuthAgentAuth", func() {
+	var (
+		token string
+		key   string
+		a     *LocalAuthenticator
+	)
+
+	BeforeEach(func() {
+		var err error
+		token, key, err = ECDSATokenAndKey()
+		Expect(err).ToNot(HaveOccurred())
+
+		cfg := &Config{ECPublicKey: key}
+
+		a, err = NewLocalAuthenticator(cfg, logrus.New())
+		Expect(err).ToNot(HaveOccurred())
+	})
+
```

Comment:
Add a unit test to make sure the code verifies that the correct signing method is used

Reply:
I'm actually having a hard time creating a test for this.
If I use a valid token of a different type or edit the header to list a different value it just fails the parse rather than our conditional.

I'm thinking of refactoring that method to set the [`ValidMethods` list in the parser](https://github.com/dgrijalva/jwt-go/blob/317b82a6816a5f2b31ff8176cd6c7e5fe7564093/parser.go#L11) and let the library do that validation for us.

I'll still add a test with a different key type though.

Reply:
> it just fails the parse rather than our conditional.
That's fine, as long as something is failing. I just wish there's some test to remind us of this issue if someone someday decides to refactor your code / use a different library that doesn't protect you

Reply:
Let me know if you think the new tests I added don't handle the cases you have in mind.
https://github.com/openshift/assisted-service/pull/1228/files#diff-9346320c59d8ca28654cfb3cc144c048368b13ee7d39ec00018753b1e1663c75R67-R77